crypto/types.go

```diff
@@ -117,8 +117,17 @@ type Signature []byte
 // invalid inputs.
 type invalidInputsError struct {
 	error
+	subType invalidInputErrorSubType
 }
 
+type invalidInputErrorSubType int
+
+const (
+	Other invalidInputErrorSubType = 0
+	dkgInvalidStateTransition
+	aggregationEmptyList
```

Comment:
better to define a `Error` method, that include the error type name in the error message, so that we can see clear error type in the error message rather than 1, 2.

Reply:
That's how I started to implement it. However I wanted the new errors to be "subtypes" of the existing sentinel error `invalidInputsError` , i.e, I want this code to pass:
```
subTyepeError := aggregationEmptyListErrorf("some error")
assert.True(t, IsAggregationEmptyListError(subTyepeError))
assert.True(t, IsInvalidInputsError(subTyepeError))
```
If I declare errors as a separate error type, the last assertion would fail, because `errors.As` fails. I didn't find a better design to achieve that.
Regarding understanding the error message, it won't be necessary to look at the number because the text should be self-explanatory.

Reply:
I see. Yeah, I like your solution. Simple.

But for your reference, this is how to declare errors as a separate error type, the benefit is we can get rid of the `generic` sub type.

https://go.dev/play/p/DOrPlVOEZ1m

Reply:
Ah sweet, that's exactly how I implemented it first, but forgot the `Unwrap` function and it didn't work, https://go.dev/play/p/N7xniNVRH3i. I like your solution better, I might switch to it!